typescript/cli/src/config/agent.ts

```diff
@@ -1,36 +1,64 @@
+import { confirm } from '@inquirer/prompts';
 import { fromError } from 'zod-validation-error';
 
+import { ChainAddresses } from '@hyperlane-xyz/registry';
 import {
   AgentConfigSchema,
   ChainMap,
   HyperlaneCore,
   HyperlaneDeploymentArtifacts,
   buildAgentConfig,
 } from '@hyperlane-xyz/sdk';
-import { objMap, promiseObjAll } from '@hyperlane-xyz/utils';
+import { objFilter, objMap, promiseObjAll } from '@hyperlane-xyz/utils';
 
 import { CommandContext } from '../context/types.js';
-import { logBlue, logGreen, logRed } from '../logger.js';
+import { errorRed, logBlue, logGreen, warnYellow } from '../logger.js';
 import { writeYamlOrJson } from '../utils/files.js';
 
 export async function createAgentConfig({
   context,
   chains,
   out,
+  skipPrompts = false,
 }: {
   context: CommandContext;
-  chains: string[];
+  chains?: string[];
   out: string;
+  skipPrompts?: boolean;
 }) {
   logBlue('\nCreating agent config...');
 
   const { registry, multiProvider, chainMetadata } = context;
   const addresses = await registry.getAddresses();
 
-  const core = HyperlaneCore.fromAddressesMap(addresses, multiProvider);
+  if (!chains) {
+    if (skipPrompts) {
```

Comment:
IIRC @yorhodes and @nbayindirli had some kind of new utility fn for this pattern of 'prompt if not provided'

Reply:
`export async function detectAndConfirmOrPrompt(
  detect: () => Promise<string | undefined>,
  prompt: string,
  label: string,
  source?: string,
): Promise<string> {
  let detectedValue: string | undefined;
  try {
    detectedValue = await detect();
    if (detectedValue) {
      const confirmed = await confirm({
        message: `Detected ${label} as ${detectedValue}${
          source ? ` from ${source}` : ''
        }, is this correct?`,
      });
      if (confirmed) {
        return detectedValue;
      }
    }
    // eslint-disable-next-line no-empty
  } catch (e) {}
  return input({ message: `${prompt} ${label}:`, default: detectedValue });
}`

I think you are referring to this method. I'm not sure if this use case is the right fit. 

Here we infer from the absence of chains, that all chains should be used and we have the user confirm (y/n) instead of prompting them for an input.

I'm open to having the users pass in a explicit option for all chains if that feels like a better pattern. Thoughts @nbayindirli @yorhodes?

Reply:
fwiw whatever we do go with, it's important that it doesn't require interaction as we'll be running this in CI